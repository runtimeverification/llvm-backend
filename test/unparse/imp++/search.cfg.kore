// RUN: %kprint-check
/* Spa */
\or{SortGeneratedTopCell{}}(
    /* Spa */
    \or{SortGeneratedTopCell{}}(
        /* Spa */
        \equals{SortGeneratedTopCell{}, SortGeneratedTopCell{}}(
            /* Fl Fn D Sfa */ VarResult:SortGeneratedTopCell{},
            /* Fl Fn D Sfa Cl */
            Lbl'-LT-'generatedTop'-GT-'{}(
                /* Fl Fn D Sfa Cl */
                Lbl'-LT-'T'-GT-'{}(
                    /* Fl Fn D Sfa Cl */
                    Lbl'-LT-'k'-GT-'{}(
                        /* Fl Fn D Sfa Cl */
                        kseq{}(
                            /* Fl Fn D Sfa Cli */
                            /* Inj: */ inj{SortAExp{}, SortKItem{}}(
                                /* Fl Fn D Sfa Cl */
                                Lbl'PlusPlusUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'Id{}(
                                    /* Fl Fn D Sfa Cl */
                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "x")
                                )
                            ),
                            /* Fl Fn D Sfa Cl */
                            kseq{}(
                                /* Fl Fn D Sfa Cl */
                                Lbl'Hash'freezer'UndsSlshUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'AExp'Unds'AExp0'Unds'{}(
                                    /* Fl Fn D Sfa Cl */
                                    kseq{}(
                                        /* Fl Fn D Sfa Cli */
                                        /* Inj: */ inj{SortAExp{}, SortKItem{}}(
                                            /* Fl Fn D Sfa Cl */
                                            Lbl'UndsSlshUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'AExp'Unds'AExp{}(
                                                /* Fl Fn D Sfa Cl */
                                                Lbl'PlusPlusUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'Id{}(
                                                    /* Fl Fn D Sfa Cl */
                                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                                    "x")
                                                ),
                                                /* Fl Fn D Sfa Cli */
                                                /* Inj: */ inj{SortId{}, SortAExp{}}(
                                                    /* Fl Fn D Sfa Cl */
                                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                                    "x")
                                                )
                                            )
                                        ),
                                        /* Fl Fn D Sfa Cl */ dotk{}()
                                    )
                                ),
                                /* Fl Fn D Sfa Cl */
                                kseq{}(
                                    /* Fl Fn D Sfa Cl */
                                    Lbl'Hash'freezer'UndsEqlsUndsSClnUnds'IMP-SYNTAX'Unds'Stmt'Unds'Id'Unds'AExp1'Unds'{}(
                                        /* Fl Fn D Sfa Cl */
                                        kseq{}(
                                            /* Fl Fn D Sfa Cli */
                                            /* Inj: */ inj{SortId{}, SortKItem{}}(
                                                /* Fl Fn D Sfa Cl */
                                                \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                                "y")
                                            ),
                                            /* Fl Fn D Sfa Cl */ dotk{}()
                                        )
                                    ),
                                    /* Fl Fn D Sfa Cl */ dotk{}()
                                )
                            )
                        )
                    ),
                    /* Fl Fn D Sfa Cl */
                    Lbl'-LT-'state'-GT-'{}(
                        /* Fl Fn D Sfa Cl */
                        /* builtin: */
                        Lbl'Unds'Map'Unds'{}(
                            /* concrete element: */ Lbl'UndsPipe'-'-GT-Unds'{}(
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortId{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "x")
                                ),
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortInt{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    /* builtin: */ \dv{SortInt{}}("1")
                                )
                            ),
                            /* concrete element: */ Lbl'UndsPipe'-'-GT-Unds'{}(
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortId{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "y")
                                ),
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortInt{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    /* builtin: */ \dv{SortInt{}}("0")
                                )
                            )
                        )
                    )
                ),
                /* Fl Fn D Sfa Cl */
                Lbl'-LT-'generatedCounter'-GT-'{}(
                    /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
                )
            )
        ),
        /* Spa */
        \equals{SortGeneratedTopCell{}, SortGeneratedTopCell{}}(
            /* Fl Fn D Sfa */ VarResult:SortGeneratedTopCell{},
            /* Fl Fn D Sfa Cl */
            Lbl'-LT-'generatedTop'-GT-'{}(
                /* Fl Fn D Sfa Cl */
                Lbl'-LT-'T'-GT-'{}(
                    /* Fl Fn D Sfa Cl */
                    Lbl'-LT-'k'-GT-'{}(
                        /* Fl Fn D Sfa Cl */
                        kseq{}(
                            /* Fl Fn D Sfa Cli */
                            /* Inj: */ inj{SortAExp{}, SortKItem{}}(
                                /* Fl Fn D Sfa Cl */
                                Lbl'PlusPlusUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'Id{}(
                                    /* Fl Fn D Sfa Cl */
                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "x")
                                )
                            ),
                            /* Fl Fn D Sfa Cl */
                            kseq{}(
                                /* Fl Fn D Sfa Cl */
                                Lbl'Hash'freezer'UndsSlshUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'AExp'Unds'AExp0'Unds'{}(
                                    /* Fl Fn D Sfa Cl */
                                    kseq{}(
                                        /* Fl Fn D Sfa Cli */
                                        /* Inj: */ inj{SortId{}, SortKItem{}}(
                                            /* Fl Fn D Sfa Cl */
                                            \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                            "x")
                                        ),
                                        /* Fl Fn D Sfa Cl */ dotk{}()
                                    )
                                ),
                                /* Fl Fn D Sfa Cl */
                                kseq{}(
                                    /* Fl Fn D Sfa Cl */
                                    Lbl'Hash'freezer'UndsSlshUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'AExp'Unds'AExp1'Unds'{}(
                                        /* Fl Fn D Sfa Cl */
                                        kseq{}(
                                            /* Fl Fn D Sfa Cli */
                                            /* Inj: */ inj{SortAExp{}, SortKItem{}}(
                                                /* Fl Fn D Sfa Cl */
                                                Lbl'PlusPlusUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'Id{}(
                                                    /* Fl Fn D Sfa Cl */
                                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                                    "x")
                                                )
                                            ),
                                            /* Fl Fn D Sfa Cl */ dotk{}()
                                        )
                                    ),
                                    /* Fl Fn D Sfa Cl */
                                    kseq{}(
                                        /* Fl Fn D Sfa Cl */
                                        Lbl'Hash'freezer'UndsEqlsUndsSClnUnds'IMP-SYNTAX'Unds'Stmt'Unds'Id'Unds'AExp1'Unds'{}(
                                            /* Fl Fn D Sfa Cl */
                                            kseq{}(
                                                /* Fl Fn D Sfa Cli */
                                                /* Inj: */ inj{SortId{}, SortKItem{}}(
                                                    /* Fl Fn D Sfa Cl */
                                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                                    "y")
                                                ),
                                                /* Fl Fn D Sfa Cl */ dotk{}()
                                            )
                                        ),
                                        /* Fl Fn D Sfa Cl */ dotk{}()
                                    )
                                )
                            )
                        )
                    ),
                    /* Fl Fn D Sfa Cl */
                    Lbl'-LT-'state'-GT-'{}(
                        /* Fl Fn D Sfa Cl */
                        /* builtin: */
                        Lbl'Unds'Map'Unds'{}(
                            /* concrete element: */ Lbl'UndsPipe'-'-GT-Unds'{}(
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortId{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "x")
                                ),
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortInt{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    /* builtin: */ \dv{SortInt{}}("1")
                                )
                            ),
                            /* concrete element: */ Lbl'UndsPipe'-'-GT-Unds'{}(
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortId{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "y")
                                ),
                                /* Fl Fn D Sfa Cli */
                                /* Inj: */ inj{SortInt{}, SortKItem{}}(
                                    /* Fl Fn D Sfa Cl */
                                    /* builtin: */ \dv{SortInt{}}("0")
                                )
                            )
                        )
                    )
                ),
                /* Fl Fn D Sfa Cl */
                Lbl'-LT-'generatedCounter'-GT-'{}(
                    /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
                )
            )
        )
    ),
    /* Spa */
    \equals{SortGeneratedTopCell{}, SortGeneratedTopCell{}}(
        /* Fl Fn D Sfa */ VarResult:SortGeneratedTopCell{},
        /* Fl Fn D Sfa Cl */
        Lbl'-LT-'generatedTop'-GT-'{}(
            /* Fl Fn D Sfa Cl */
            Lbl'-LT-'T'-GT-'{}(
                /* Fl Fn D Sfa Cl */
                Lbl'-LT-'k'-GT-'{}(
                    /* Fl Fn D Sfa Cl */
                    kseq{}(
                        /* Fl Fn D Sfa Cli */
                        /* Inj: */ inj{SortAExp{}, SortKItem{}}(
                            /* Fl Fn D Sfa Cl */
                            Lbl'PlusPlusUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'Id{}(
                                /* Fl Fn D Sfa Cl */
                                \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "x")
                            )
                        ),
                        /* Fl Fn D Sfa Cl */
                        kseq{}(
                            /* Fl Fn D Sfa Cl */
                            Lbl'Hash'freezer'UndsSlshUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'AExp'Unds'AExp0'Unds'{}(
                                /* Fl Fn D Sfa Cl */
                                kseq{}(
                                    /* Fl Fn D Sfa Cli */
                                    /* Inj: */ inj{SortInt{}, SortKItem{}}(
                                        /* Fl Fn D Sfa Cl */
                                        /* builtin: */ \dv{SortInt{}}("1")
                                    ),
                                    /* Fl Fn D Sfa Cl */ dotk{}()
                                )
                            ),
                            /* Fl Fn D Sfa Cl */
                            kseq{}(
                                /* Fl Fn D Sfa Cl */
                                Lbl'Hash'freezer'UndsSlshUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'AExp'Unds'AExp1'Unds'{}(
                                    /* Fl Fn D Sfa Cl */
                                    kseq{}(
                                        /* Fl Fn D Sfa Cli */
                                        /* Inj: */ inj{SortAExp{}, SortKItem{}}(
                                            /* Fl Fn D Sfa Cl */
                                            Lbl'PlusPlusUndsUnds'IMP-SYNTAX'Unds'AExp'Unds'Id{}(
                                                /* Fl Fn D Sfa Cl */
                                                \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                                "x")
                                            )
                                        ),
                                        /* Fl Fn D Sfa Cl */ dotk{}()
                                    )
                                ),
                                /* Fl Fn D Sfa Cl */
                                kseq{}(
                                    /* Fl Fn D Sfa Cl */
                                    Lbl'Hash'freezer'UndsEqlsUndsSClnUnds'IMP-SYNTAX'Unds'Stmt'Unds'Id'Unds'AExp1'Unds'{}(
                                        /* Fl Fn D Sfa Cl */
                                        kseq{}(
                                            /* Fl Fn D Sfa Cli */
                                            /* Inj: */ inj{SortId{}, SortKItem{}}(
                                                /* Fl Fn D Sfa Cl */
                                                \dv{SortId{}}(/* Fl Fn D Sfa Cl */
                                                "y")
                                            ),
                                            /* Fl Fn D Sfa Cl */ dotk{}()
                                        )
                                    ),
                                    /* Fl Fn D Sfa Cl */ dotk{}()
                                )
                            )
                        )
                    )
                ),
                /* Fl Fn D Sfa Cl */
                Lbl'-LT-'state'-GT-'{}(
                    /* Fl Fn D Sfa Cl */
                    /* builtin: */
                    Lbl'Unds'Map'Unds'{}(
                        /* concrete element: */ Lbl'UndsPipe'-'-GT-Unds'{}(
                            /* Fl Fn D Sfa Cli */
                            /* Inj: */ inj{SortId{}, SortKItem{}}(
                                /* Fl Fn D Sfa Cl */
                                \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "x")
                            ),
                            /* Fl Fn D Sfa Cli */
                            /* Inj: */ inj{SortInt{}, SortKItem{}}(
                                /* Fl Fn D Sfa Cl */
                                /* builtin: */ \dv{SortInt{}}("1")
                            )
                        ),
                        /* concrete element: */ Lbl'UndsPipe'-'-GT-Unds'{}(
                            /* Fl Fn D Sfa Cli */
                            /* Inj: */ inj{SortId{}, SortKItem{}}(
                                /* Fl Fn D Sfa Cl */
                                \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "y")
                            ),
                            /* Fl Fn D Sfa Cli */
                            /* Inj: */ inj{SortInt{}, SortKItem{}}(
                                /* Fl Fn D Sfa Cl */
                                /* builtin: */ \dv{SortInt{}}("0")
                            )
                        )
                    )
                )
            ),
            /* Fl Fn D Sfa Cl */
            Lbl'-LT-'generatedCounter'-GT-'{}(
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
            )
        )
    )
)
