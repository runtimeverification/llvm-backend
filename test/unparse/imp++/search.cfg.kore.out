  {
    Result:GeneratedTopCell
  #Equals
    [38;5;226m<T>[0m
      [38;5;46m<k>[0m
        ++[0m x ~> #freezer_/__IMP-SYNTAX_AExp_AExp_AExp0_[0m ([0m 1 ~> .K )[0m ~> #freezer_/__IMP-SYNTAX_AExp_AExp_AExp1_[0m ([0m ++[0m x ~> .K )[0m ~> #freezer_=_;_IMP-SYNTAX_Stmt_Id_AExp1_[0m ([0m y ~> .K )[0m ~> .K
      [38;5;46m</k>[0m
      [38;5;196m<state>[0m
        x |->[0m 1
        y |->[0m 0
      [38;5;196m</state>[0m
    [38;5;226m</T>[0m
  }
#Or
  {
    Result:GeneratedTopCell
  #Equals
    [38;5;226m<T>[0m
      [38;5;46m<k>[0m
        ++[0m x ~> #freezer_/__IMP-SYNTAX_AExp_AExp_AExp0_[0m ([0m ++[0m x /[0m x ~> .K )[0m ~> #freezer_=_;_IMP-SYNTAX_Stmt_Id_AExp1_[0m ([0m y ~> .K )[0m ~> .K
      [38;5;46m</k>[0m
      [38;5;196m<state>[0m
        x |->[0m 1
        y |->[0m 0
      [38;5;196m</state>[0m
    [38;5;226m</T>[0m
  }
#Or
  {
    Result:GeneratedTopCell
  #Equals
    [38;5;226m<T>[0m
      [38;5;46m<k>[0m
        ++[0m x ~> #freezer_/__IMP-SYNTAX_AExp_AExp_AExp0_[0m ([0m x ~> .K )[0m ~> #freezer_/__IMP-SYNTAX_AExp_AExp_AExp1_[0m ([0m ++[0m x ~> .K )[0m ~> #freezer_=_;_IMP-SYNTAX_Stmt_Id_AExp1_[0m ([0m y ~> .K )[0m ~> .K
      [38;5;46m</k>[0m
      [38;5;196m<state>[0m
        x |->[0m 1
        y |->[0m 0
      [38;5;196m</state>[0m
    [38;5;226m</T>[0m
  }
